src/session.ts

```diff
@@ -146,7 +150,13 @@ export const createState = (initial: State) => {
       store.update(s => {
         switch (s.connection) {
           case Connection.Connected:
-            s.session.address = address; 
+            // In case of locking Metamask the accountsChanged event returns undefined.
+            // To prevent out of sync state, the wallet gets disconnected.
+            if(address === undefined) disconnectWallet();
```

Comment:
Little style thing: we usually have whitespace between keywords and the `()`, eg. `if (...)`, `function (...)`, etc., same on line 36.